### DISCUSSION TITLE
OOXML Strict/Transitional

### DISCUSSION THREAD

**Discussion by Asbjoedt** [mikeebowen/ooxml-validator-vscode#13] [Q&A] [answered]
Hi

Can the validator differentiate between and validate OOXML Strict and OOXML Transitional, so that I know which one of the latter my file is and whether it is valid/well-formed according to strict or transitional rules?

**Comment by mikeebowen** [comment, marked as answer]
Hi @Asbjoedt ,

Currently the validator doesn't distinguish between OOXML Strict and Transitional. It could be possible; it depends on whether or not the validation API from the [Open Xml SDK](https://github.com/OfficeDev/Open-XML-SDK) can make that distinction. 

The validator works by using .Net runtime to call a [.Net Project](https://github.com/mikeebowen/OOXML-Validator) that uses the Open Xml SDK's validation to validate the file. If the SDK can make that distinction, then it could be added to the return data to the validator and displayed with the other errors.

I don't know if the Open Xml SDK can make the distinction and, unfortunately, I don't have a lot of bandwidth right now to work on this. But if you can do some research or even better, submit a pull request, I'd be happy to add this functionality.

**Reply by Asbjoedt** [reply]
Thanks for replying @mikeebowen 

_Our motivation
I work for a national archive and we are very interested in seeing the Strict variant promoted and more widely adopted by Excel and its supporting tools for validation and automatic worksheet manipulation. If so, we can adopt Strict as our preservation file format and avoid having to migrate our data collections of spreadsheets from Transitional to Strict some time in the future, but instead do it upfront before or during submission/ingest of data._

I did a tiny bit of research and found this:
https://github.com/OfficeDev/Open-XML-SDK/pull/716
https://github.com/OfficeDev/Open-XML-SDK/issues/711

If the commit is merged, would this provide your necessary framework?

As is, I cannot provide any pull requests unfortunately.

**Reply by mikeebowen** [reply]
Hi @Asbjoedt, that PR was merged in the SDK, so we are able to determine strict vs transitional. Most of the changes will actually be in the [OOXML Validator](https://github.com/mikeebowen/OOXML-Validator). This isn't a validation error, so I will have to extend the validator and get the information manually and add it in. I will work to get this in when I can. Please create a feature request on the [issues page](https://github.com/mikeebowen/ooxml-validator-vscode/issues) to help track this. Thanks!

**Reply by Asbjoedt** [reply]
That is great news. I couldn't see it was merged.
I will create the feature request and track it. Reach out if you need any help other than coding.

**Reply by mikeebowen** [reply]
Since the PR was merged there has been a lot of refactoring, but I used the debugger to verify that it was available and public. In the code it looks like only the get is public, so we'll be able to read it, but not change the value.
